pkg/cloud/services/network/subnets.go

```diff
@@ -399,6 +410,15 @@ func (s *Service) describeSubnets() (*ec2.DescribeSubnetsOutput, error) {
 }
 
 func (s *Service) createSubnet(sn *infrav1.SubnetSpec) (*infrav1.SubnetSpec, error) {
+	// When managing subnets, the ID specified in the spec is the name of the subnet.
+	if sn.Tags == nil {
+		sn.Tags = make(infrav1.Tags)
+	}
+	if len(sn.ID) > 0 {
+		sn.Tags["Name"] = sn.ID
+	}
```

Comment:
Please deep-copy the `Tags` field into a local variable so we don't introduce strange bugs by changing the input parameter. Also, the `Name` tag may already be given and must be respected, so the `id` field should be the fallback value if _no_ `Name` tag is specified.

Reply:
Agreed on the latter point about fallback, although not sure why we'd need to deepcopy the tags given that we want to change the input?